### ISSUE TITLE
Training diverges with unchanged cocoft_swint.py config

### ISSUE THREAD

**Issue by winnietaoxinyi** [Christinepan881/DINO-R1#4]
Running the provided code on COCO2017 with tools/dist_train.sh leads to unstable training from epoch 1, specifically NaN for loss and all grpo_loss and large grad_norm. I manually put all the DINO R1 code inside the mmdetection and did not change anything else. Below are one line from the output:

`{"base_lr": 0.0003999999999999992, "lr": 0.0003999999999999992, "data_time": 0.021377196311950685, "grad_norm": 1051.5165881347657, "loss": NaN, "loss_cls": 0.5884847468137742, "loss_bbox": 0.25503098115324974, "loss_iou": 0.5586857917904854, "d0.loss_cls": 0.6208624422550202, "d0.loss_bbox": 0.27402777165174486, "d0.loss_iou": 0.5838584011793136, "d1.loss_cls": 0.6116782647371292, "d1.loss_bbox": 0.26108846336603164, "d1.loss_iou": 0.5674582782387734, "d2.loss_cls": 0.5935451281070709, "d2.loss_bbox": 0.2575897231698036, "d2.loss_iou": 0.5646088269352912, "d3.loss_cls": 0.5895848894119262, "d3.loss_bbox": 0.25648326605558397, "d3.loss_iou": 0.5613557940721512, "d4.loss_cls": 0.5886811619997024, "d4.loss_bbox": 0.255378295481205, "d4.loss_iou": 0.5591926276683807, "enc_loss_cls": 0.6337313371896743, "enc_loss_bbox": 0.32812991589307783, "enc_loss_iou": 0.667872279882431, "dn_loss_cls": 0.3326878523826599, "dn_loss_bbox": 0.5274576833844185, "dn_loss_iou": 0.7123418909311294, "d0.dn_loss_cls": 0.35078244805336, "d0.dn_loss_bbox": 0.6206773555278778, "d0.dn_loss_iou": 0.8400408971309662, "d1.dn_loss_cls": 0.3410477802157402, "d1.dn_loss_bbox": 0.5497661462426185, "d1.dn_loss_iou": 0.74714200258255, "d2.dn_loss_cls": 0.33633452743291853, "d2.dn_loss_bbox": 0.5314363485574722, "d2.dn_loss_iou": 0.7208197140693664, "d3.dn_loss_cls": 0.3331788030266762, "d3.dn_loss_bbox": 0.5287106049060821, "d3.dn_loss_iou": 0.7149484157562256, "d4.dn_loss_cls": 0.3333020284771919, "d4.dn_loss_bbox": 0.5274295264482498, "d4.dn_loss_iou": 0.7122027957439423, "grpo_loss_0": NaN, "grpo_loss_1": NaN, "grpo_loss_2": NaN, "grpo_loss_3": NaN, "grpo_loss_4": NaN, "grpo_loss_5": NaN, "lv_contra_loss": 174.1887907409668, "time": 6.798497500419617, "epoch": 1, "iter": 3600, "memory": 35182, "step": 3600}`

Could you share more code or instructions in order to reproduce the results in the paper— like LR/weight warmup (including GRPO/contrastive weights), package versions (torch/mmcv/mmengine/fairscale), and any hardware/training settings (batch size, num negatives, temperature)? Thanks.

**Comment by KeepLost** [comment]
Current code is insufficient to integrate into mmdetection, because the `DINO_R1_Head.grpo_loss` method calls `self.assigner.get_grpo_advantage` method, which is not implemented in any assigner class. I tried using ChatGPT to supplement this implementation, but got NaN loss like what you see in your log output, and my `grad_norm` also gets NaN.
